subsys/bluetooth/host/classic/sco.c

```diff
@@ -0,0 +1,409 @@
+/* sco.c - Bluetooth sco handling */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/check.h>
+
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include "common/bt_str.h"
+
+#include "host/addr_internal.h"
+#include "host/hci_core.h"
+#include "br.h"
+#include "host/conn_internal.h"
+#include "sco_internal.h"
+
+#define LOG_LEVEL CONFIG_BT_CONN_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_sco);
+
+struct bt_sco_server *sco_server;
+
+#define SCO_CHAN(_sco) ((_sco)->sco.chan);
+
+int bt_sco_server_register(struct bt_sco_server *server)
+{
+	CHECKIF(!server) {
+		LOG_DBG("Invalid parameter: server %p", server);
+		return -EINVAL;
+	}
+
+	if (sco_server) {
+		return -EADDRINUSE;
+	}
+
+	if (!server->accept) {
+		return -EINVAL;
+	}
+
+#if defined(CONFIG_BT_SMP)
+	if (server->sec_level > BT_SECURITY_L3) {
+		return -EINVAL;
+	}
+#endif /* CONFIG_BT_SMP */
```

Comment:
This part doesn't make sense to me. BR/EDR will in practice always have security support available, and it's not dependent on whether SMP is available (which is generally only used for LE).

Reply:
L2CAP fixed channel of BR/EDR Security Manager is also implemented in `smp.c`. The file `smp.c` depends on `BT_SMP`. In current Kconfig, `BT_SMP` is selected by `BT_CLASSIC`. 
If the preprocessor directive `#if defined(CONFIG_BT_SMP)` ... `#endif /* CONFIG_BT_SMP */` is removed, it has no impact on compilation. But from the dependency perspective, the preprocessor directive `#if defined(CONFIG_BT_SMP)` ... `#endif /* CONFIG_BT_SMP */` is required.

Reply:
The BR/EDR SMP channel exists only for cross-transport key derivation. The actual BR/EDR pairing happens using BR/EDR-only methods (i.e. through Secure Simple Pairing or legacy BR/EDR pairing). The ability to establish a secure connection for BR/EDR, or the existence of a minimum security level requirement do not depend on SMP, so it looks illogical to have such dependencies in the code.

Whatever the underlying cause is (since I didn't completely understand it) for you wanting to have these #ifdefs should be fixed instead.

Reply:
I think the preprocessor directive can be removed for BR/EDR. I will update it.